### DISCUSSION TITLE
How does it work? db.commit() db.refresh()

### DISCUSSION THREAD

**Discussion by tranvannhat** [fastapi/fastapi#8917] [Questions]
In 1 function I call
(1) `crud.user_work_history.create(db, obj_in=user_work_history_in)`
... and then call
(2) `crud.user.update_when_leave_work(db, db_obj=user_leave, obj_in=user_in)`

but 1 working, 2 working but not update new data. Please help me.
How does it work db.commit() db.refresh(db_obj)?

**Comment by krishnardt** [comment]
Hi,

this issue is related to Sqlalchemy ORM and the database. 
you can check or post in their respective pages.

For information about commit and refresh, look into below examples. They may help you.

[https://stackoverflow.com/questions/4201455/sqlalchemy-whats-the-difference-between-flush-and-commit](https://stackoverflow.com/questions/4201455/sqlalchemy-whats-the-difference-between-flush-and-commit
)
[https://michaelcho.me/article/sqlalchemy-commit-flush-expire-refresh-merge-whats-the-difference](https://michaelcho.me/article/sqlalchemy-commit-flush-expire-refresh-merge-whats-the-difference)
